stubs/Pillow/PIL/Image.pyi

```diff
@@ -143,39 +147,37 @@ class Image:
     ) -> Image: ...
     def copy(self) -> Image: ...
     __copy__ = copy
-    def crop(self, box=...) -> Image: ...
-    def draft(self, mode, size) -> None: ...
-    def filter(self, filter): ...
+    def crop(self, box: _Box | None = ...) -> Image: ...
+    def draft(self, mode: str, size: _Size) -> None: ...
+    def filter(self, filter: Filter | Callable[[], Filter]) -> Image: ...
     def getbands(self) -> Tuple[str, ...]: ...
     def getbbox(self) -> tuple[int, int, int, int] | None: ...
-    def getcolors(self, maxcolors: int = ...): ...
-    def getdata(self, band=...): ...
+    def getcolors(self, maxcolors: int = ...) -> list[tuple[int, int]]: ...
+    def getdata(self, band: int | None = ...): ...
     def getextrema(self): ...
     def getexif(self) -> Exif: ...
     def getim(self): ...
-    def getpalette(self): ...
-    def getpixel(self, xy): ...
-    def getprojection(self): ...
-    def histogram(self, mask=..., extrema=...): ...
-    def entropy(self, mask=..., extrema=...): ...
-    def paste(
-        self, im: Image, box: tuple[float, float] | tuple[float, float, float, float] | None = ..., mask: Image | None = ...
-    ) -> None: ...
+    def getpalette(self) -> list[int] | None: ...
+    def getpixel(self, xy: tuple[int, int]): ...
+    def getprojection(self) -> tuple[list[int], list[int]]: ...
+    def histogram(self, mask: Image | None = ..., extrema: tuple[int, int] | tuple[float, float] | None = ...) -> list[int]: ...
+    def entropy(self, mask: Image | None = ..., extrema: tuple[int, int] | tuple[float, float] | None = ...) -> float: ...
+    def paste(self, im: Image, box: tuple[float, float] | _Box | None = ..., mask: Image | None = ...) -> None: ...
     def alpha_composite(self, im: Image, dest: tuple[int, int] = ..., source: tuple[int, int] = ...) -> None: ...
-    def point(self, lut, mode=...): ...
-    def putalpha(self, alpha): ...
-    def putdata(self, data, scale: float = ..., offset: float = ...): ...
-    def putpalette(self, data, rawmode=...): ...
-    def putpixel(self, xy, value): ...
-    def remap_palette(self, dest_map, source_palette=...): ...
+    def point(self, lut, mode: str | None = ...) -> Image: ...
+    def putalpha(self, alpha: Image | int) -> None: ...
+    def putdata(self, data: Sequence[int], scale: float = ..., offset: float = ...) -> None: ...
+    def putpalette(self, data: ImagePalette | bytes | str | Iterable[int] | SupportsBytes, rawmode: str | None = ...) -> None: ...
```

Comment:
I don't think str works (https://github.com/python-pillow/Pillow/blob/master/src/PIL/Image.py#L1698). The code suggests you need ImagePalette, bytes, or SupportsBytes.

Reply:
Not sure I agree; it calls `bytes(data)`, so it should be able to be anything that can be passed to `bytes`'s init.

https://github.com/python-pillow/Pillow/blob/bdc87f6dcf97fd4851732e2b3bfaade72b060773/src/PIL/Image.py#L1768

Reply:
(In fact, the docstring says "or string".)

Reply:
But `bytes("x")` throws an exception in Python 3. I feel like that docstring may be a Python 2 remnant.

Reply:
Hm, I guess that's right; I missed when copying the types from https://github.com/python/typeshed/blob/14d7431ff1f554ae1d103b996995bfe3ecec67bd/stdlib/builtins.pyi#L420 that you need to give it an encoding.

Reply:
The types for bytes recently changed, maybe this needs to be:

```
ImagePalette | bytes | Iterable[SupportsIndex] | SupportsBytes
```

to match `bytes`? You said `ImagePalette | bytes | SupportsBytes`, but my thought would be to keep it close to `bytes`.

Reply:
Yes, that seems right. I assumed that `SupportsBytes` is the same as what bytes accepts but should have checked.

Reply:
Yeah, I was surprised too.

Reply:
Removed `str` (and fixed the other stuff).